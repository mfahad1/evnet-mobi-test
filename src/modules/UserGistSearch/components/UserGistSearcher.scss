.UserGistSearcher {
  display: flex;
  align-items: center;
  flex-direction: column;
}

.UserGistSearcher_Form {
  display: flex;
  align-items: center;

  .Input {
    padding: 20px;
    margin: 10px 0;
    padding: 10px;
    border: 0;
    box-shadow: 0 0 15px 4px rgba(0, 0, 0, 0.06);
    padding: 10px;
    border-radius: 10px;
    font-size: 2rem;

    :focus {
      outline: none !important;
      border-color: #719ECE;
      box-shadow: 0 0 10px #719ECE;
    }
  }
}

.UserGistSearcher_Button {
  background-color: transparent;
  width: 3rem;

  cursor: pointer;
  border: none;
  padding: 0;
  margin-left: 1rem;

  :hover {
    border-radius: 0.5rem;
  }

  :focus {
    outline: none;
    box-shadow: 0 0 0 3px lightskyblue;
  }

  :active {
    background-color: white;
    border-color: #333;
  }
}