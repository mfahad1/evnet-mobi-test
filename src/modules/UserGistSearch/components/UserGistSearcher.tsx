import InputField from '../../../components/InputField';
import { ReactComponent as SearchSVG } from '../../../resources/icons/search.svg';
import './UserGistSearcher.scss';

type UserGistSearcherProps = {
  onComplete: (val: string) => void;
};

export default function UserGistSearcher({ onComplete }: UserGistSearcherProps) {
  return (
    <div className="UserGistSearcher">
      <h1>Search Git Gist By User Name</h1>
      <div className="UserGistSearcher_Form">
        <InputField
          onComplete={onComplete}
          debounce={1000}
          onEnter={onComplete}
          className="Input"
          placeholder="Enter Username"
        />

        <button className="UserGistSearcher_Button">
          <SearchSVG />
        </button>
      </div>
    </div>
  );
}
