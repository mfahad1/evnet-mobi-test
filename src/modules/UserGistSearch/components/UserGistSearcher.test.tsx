import renderer from 'react-test-renderer';
import UserGistSearcher from './UserGistSearcher';

describe('WithLoader', () => {
  it('should match snapshot', () => {
    const elem = renderer.create(<UserGistSearcher onComplete={jest.fn} />).toJSON();

    expect(elem).toMatchSnapshot();
  });
});
