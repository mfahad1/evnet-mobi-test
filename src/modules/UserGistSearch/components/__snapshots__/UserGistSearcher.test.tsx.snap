// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithLoader should match snapshot 1`] = `
<div
  className="UserGistSearcher"
>
  <h1>
    Search Git Gist By User Name
  </h1>
  <div
    className="UserGistSearcher_Form"
  >
    <input
      className="Input"
      data-testid="InputField"
      onChange={[Function]}
      onKeyUp={[Function]}
      placeholder="Enter Username"
    />
    <button
      className="UserGistSearcher_Button"
    >
      <svg>
        search.svg
      </svg>
    </button>
  </div>
</div>
`;
